### ISSUE TITLE
Error when making html in docs

### ISSUE THREAD

**Issue by nateanl** [pytorch/audio#1677]
## To Reproduce

```
cd docs
make html
```

## Expected behavior

```
reading sources... [100%] utils                                                                                                                                                     

Warning, treated as error:
error while formatting arguments for torchaudio.backend.sox_io_backend.load: type object 'tuple' has no attribute '_special'
make: *** [html] Error 2
```
## Environment

 - What commands did you used to install torchaudio (conda/pip/build from source)? build from source
 - If you are building from source, which commit is it? ``BUILD_SOX=1 MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py develop``
 - What does `torchaudio.__version__` print? (If applicable) 0.10.0a0+77a6f41

 - PyTorch Version (e.g., 1.0): 1.10.0.dev20210803
 - OS (e.g., Linux): OSX
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): 
 - Python version: 3.9.6
 - CUDA/cuDNN version: N/A
 - GPU models and configuration: N/A
 - Any other relevant information:

cc @mthrok @carolineechen @hwangjeff

**Comment by mthrok** [comment]
Are you using the exact version of sphinx listed here?

https://github.com/pytorch/audio/blob/main/docs/requirements.txt

**Comment by nateanl** [comment]
Yes, I installed them using the command ``pip install -r requirements.txt``

**Comment by mthrok** [comment]
I cannot reproduce the issue. 🧐
Can you report your env?

```
% python -m torch.utils.collect_env
Collecting environment information...
PyTorch version: 1.10.0.dev20210804
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 11.4 (x86_64)
GCC version: Could not collect
Clang version: 11.0.0 (clang-1100.0.33.16)
CMake version: version 3.20.2
Libc version: N/A

Python version: 3.8.5 (default, Sep  4 2020, 02:22:02)  [Clang 10.0.0 ] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.2
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] torch==1.10.0.dev20210804
[pip3] torchaudio==0.10.0a0+17a3bbf
[conda] blas                      1.0                         mkl
[conda] mkl                       2021.2.0           hecd8cb5_269
[conda] mkl-service               2.3.0            py38h9ed2024_1
[conda] mkl_fft                   1.3.0            py38h4a7008c_2
[conda] mkl_random                1.2.1            py38hb2f4e1b_2
[conda] numpy                     1.20.2           py38h4b4dc7a_0
[conda] numpy-base                1.20.2           py38he0bd621_0
[conda] pytorch                   1.10.0.dev20210804         py3.8_0    pytorch-nightly
[conda] pytorch-sphinx-theme      0.0.24                    dev_0    <develop>
[conda] torch                     1.10.0.dev20210619          pypi_0    pypi
[conda] torchaudio                0.10.0a0+17a3bbf           dev_0    <develop>
```

**Comment by nateanl** [comment]
Sure.
```
Collecting environment information...
PyTorch version: 1.10.0.dev20210803
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 11.5 (x86_64)
GCC version: Could not collect
Clang version: 12.0.5 (clang-1205.0.22.11)
CMake version: version 3.19.6
Libc version: N/A

Python version: 3.9.6 (default, Jul 30 2021, 09:31:09)  [Clang 10.0.0 ] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.1
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] torch==1.10.0.dev20210803
[pip3] torchaudio==0.10.0a0+77a6f41
[pip3] torchvision==0.2.2
[conda] blas                      1.0                         mkl  
[conda] mkl                       2021.3.0                 pypi_0    pypi
[conda] mkl-include               2021.3.0                 pypi_0    pypi
[conda] mkl-service               2.4.0            py39h9ed2024_0  
[conda] mkl_fft                   1.3.0            py39h4a7008c_2  
[conda] mkl_random                1.2.2            py39hb2f4e1b_0  
[conda] numpy                     1.21.1                   pypi_0    pypi
[conda] numpy-base                1.20.1           py39h585ceec_0  
[conda] pytorch                   1.10.0.dev20210803         py3.9_0    pytorch-nightly
[conda] pytorch-sphinx-theme      0.0.24                    dev_0    <develop>
[conda] torchaudio                0.10.0a0+77a6f41           dev_0    <develop>
[conda] torchvision               0.2.2                      py_3    pytorch
```

**Comment by nateanl** [comment]
This is fixed by installing ``sphinx==4.1.0``. I'm not sure why the latest sphinx throws the error though.

**Comment by mthrok** [comment]
We might need to consider using 3.5.4, following the change in the core.

https://github.com/pytorch/pytorch/pull/61601
